skbio/metadata/_interval.py

```diff
@@ -0,0 +1,274 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from ._intersection import IntervalTree
+
+
+class Interval():
```

Comment:
should this inherit from `collections.Mapping`?  because it is basically a dict with 3 additional attributes.